### PR TITLE
Add automation pallets

### PR DESCRIPTION
**Pull request by imstar15**
### Development

patch polkadot: OK

pallet-valve: compile success
pallet-xcmp-handler: compile success
oak-primitives: compile success
pallet-autmation-time: compile success
runtime-common: compile success

Add orml-asset-registry to runtime: compile success
Add orml-xtokens to runtime: compile success
Add pallet-xcmp-handler to runtime: compile success
Add pallet-autmation-time to runtime: compile success

Remove unused patch: OK

Launch chain with moonwall: OK

### Run zombie_tanssi with moonwall

- Compile rust binaries
rust version: rustc 1.75.0

```sh
cargo build --features=fast-runtime --release
```

- Install node and pnpm

```sh
sudo npm i -g pnpm
```

- Run zombie_tanssi with moonwall
Typescript tests are run with [Moonwall](https://github.com/Moonsong-Labs/moonwall). To run these you will need to have pnpm installed:

```bash
# Install moonwall
sudo npm i -g pnpm
pnpm -g i @moonwall/cli

# Create typescript-api local interfaces
cd typescript-api && pnpm i && pnpm create-local-interfaces && cd ..

# Change directory to test
cd test

# Run zombienet tests (with container-chains)
pnpm i
pnpm moonwall run zombie_tanssi
```

- Test task extrinsic

https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:9949#/explorer

https://polkadot.js.org/apps/?rpc=ws%3A%2F%2F127.0.0.1%3A9949#/extrinsics/decode/0xce030004903dcb66000000000a00008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a480b0060defb7405

image

### Unit Tests

```
cargo test --features=fast-runtime --release
```

**pallet-xcmp-handler**

```
     Running unittests src/lib.rs (/Users/star/devlab/tanssi/target/release/deps/pallet_xcmp_handler-3aac3cb4dcd5a55c)

running 10 tests
test mock::test_genesis_config_builds ... ok
test mock::__construct_runtime_integrity_test::runtime_integrity_tests ... ok
test tests::get_instruction_set_local_currency_instructions ... ok
test tests::transact_in_target_chain_with_non_reserved_currency_will_throw_unsupported_fee_payment_error ... ok
test tests::get_local_currency_instructions_works ... ok
test tests::pay_xcm_fee_works ... ok
test tests::pay_xcm_fee_keeps_wallet_alive ... ok
test tests::transact_in_local_chain_works ... ok
test tests::transact_in_target_chain_with_to_reserved_currency_works ... ok
test tests::transact_in_target_chain_works ... ok

test result: ok. 10 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

**pallet-automation-time**

```
     Running unittests src/lib.rs (/Users/star/devlab/tanssi/target/release/deps/pallet_automation_time-276347a00b02fd0a)

running 93 tests
test fees::tests::errors_when_not_enough_funds_for_fee ... ok
test fees::tests::does_not_charge_fees_when_prereq_errors ... ok
......
test result: ok. 93 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

github actions:

CI / set-tags (pull_request): Fixed
Enforce PR labels / enforce-noteworthiness-label (pull_request): Fixed
CI / check-copyright (pull_request) : Fixed
Enforce PR labels / enforce-auditability-label (pull_request) : Fixed
CI / Check with rustfmt (pull_request): Fixed
CI / check-copyright (pull_request): Fixed
CI / toml-formatting (pull_request): Fixed
CI / cargo-clippy (pull_request) : Fixed

### CODE REVIEW

File: oak-primitives/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[package]
+name = "oak-primitives"
```
Comment (chrisli30) [inline oak-primitives/Cargo.toml]: We don’t use this name `oak` any more. This needs to be "ava-protocol-primitives".

It always need to be "ava-protocol", but can’t be "ava".

File: oak-primitives/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+version = "1.0.0"
+description = "OAK Runtime primitives"
```
Comment (chrisli30) [inline oak-primitives/Cargo.toml]: Replace all "OAK" to "Ava Protocol"

File: oak-primitives/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+authors = ["OAK Developement Team"]
+license = "GPL-3.0"
+homepage = "https://oak.tech"
```
Comment (chrisli30) [inline oak-primitives/Cargo.toml]: update website

File: oak-primitives/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+[package]
+name = "oak-primitives"
+version = "1.0.0"
+description = "OAK Runtime primitives"
+authors = ["OAK Developement Team"]
```
Comment (chrisli30) [inline oak-primitives/Cargo.toml]: Ava Protocol Team

File: oak-primitives/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+license = "GPL-3.0"
+homepage = "https://oak.tech"
+repository = 'https://github.com/OAK-Foundation/OAK-blockchain'
```
Comment (chrisli30) [inline oak-primitives/Cargo.toml]: repo address is wrong.

**Review by chrisli30 — CHANGES_REQUESTED** [review]
Looks okay, but the name change comments need to be addressed.

File: pallets/automation-time/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,100 @@
+[package]
+name = "pallet-automation-time"
+description = "Pallet for scheduling and running tasks in the future."
+version = "1.0.0"
+edition = "2021"
+authors = ["Ava Protocol Team"]
+license = "GPL-3.0"
+homepage = "https://avaprotocol.org"
+repository = 'https://github.com/AvaProtocol/OAK-blockchain'
```
Comment (chrisli30) [inline pallets/automation-time/Cargo.toml]: this repo address has changed, not OAK-blockchain any more.

Please examine all other "oak" references.

**Review by chrisli30 — CHANGES_REQUESTED** [review]

File: pallets/automation-time/Cargo.toml (hunk 6 above)
Comment (imstar15) [reply pallets/automation-time/Cargo.toml]: Fixed

**Review by chrisli30 — APPROVED** [review]
Looks good to me now.

**Review by chrisli30 — CHANGES_REQUESTED** [review]
I these CI/CD tests are built by Tanssi. They need to pass for a successful merge.

CleanShot 2024-09-01 at 18 01 58

CleanShot 2024-09-01 at 18 01 10

**Comment by chrisli30** [comment]
@imstar15 the startup instructions don’t work. There’s not even moonwall in npm.
CleanShot 2024-09-01 at 19 19 12

My feedback summary:
CleanShot 2024-09-01 at 19 07 53

The trace from the last step:
CleanShot 2024-09-01 at 19 18 31

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -574,9 +574,9 @@ jobs:
         fail-fast: false
         matrix:
           chains: [
-            { chain: "stagenet_dancebox", runtime: "dancebox" },
-            { chain: "flashbox", runtime: "flashbox" },
-            { chain: "dancebox", runtime: "dancebox" },
+            # { chain: "stagenet_dancebox", runtime: "dancebox" },
+            # { chain: "flashbox", runtime: "flashbox" },
+            # { chain: "dancebox", runtime: "dancebox" },
```
Comment (chrisli30) [inline .github/workflows/release.yml:579]: A reminder to enable these tests after upgrading 900.

**Review by chrisli30 — APPROVED** [review]
Looks good to me!

**Pull request merged**